CorsixTH/Src/th_map.cpp

```diff
@@ -1226,6 +1226,43 @@ void THMap::setTemperatureDisplay(THMapTemperatureDisplay eTempDisplay)
     if (eTempDisplay < THMT_Count) m_eTempDisplay = eTempDisplay;
 }
 
+uint32_t THMap::thermalNeighbour(uint32_t &iNeighbourSum, bool canTravel, uint32_t relative_idx, THMapNode* pNode, int prevTemp) const
+{
+    int iNeighbourCount = 0;
+
+    THMapNode* pNeighbour = pNode + relative_idx;
+
+    // Ensure the neighbour is within the map bounds
+    THMapNode* pLastNode = m_pCells + m_iWidth * m_iHeight;
+    if (pNeighbour < m_pCells || pNeighbour > pLastNode) {
```

Comment:
for a 1x1 map, ``pNeightbour`` is allowed to be ``m_pCells[0]`` and ``m_pCells[1]``, which seems wrong